New Feature: Workload Specific Compliance

Springstone opened the pull request:
<!-- Thank you for submitting a Pull Request. Please fill out the template below.-->
## Overview/Summary

This pull request includes significant updates to the Azure Landing Zones (ALZ) policy and the ALZ Portal accelerator. The changes aim to provide additional compliance controls for specific workloads, which are often required by highly regulated industries. 

Major changes include:

Updates to `.github/workflows/update-portal.yml`:

* Added a new job to build initiatives from the `src/templates/initiatives.bicep` file and output to `./eslzArm/managementGroupTemplates/policyDefinitions/initiatives.json`.

Additions to `docs/wiki/ALZ-Policies-Extra.md`:

* Added a new section "ALZ Policies - Extra" that provides information on additional policies not assigned by default or covered in the core ALZ Policies documentation. The section includes a list of policies and their descriptions.
* Introduced a new "Workload Specific Compliance" section that provides additional initiatives to enhance the security and compliance posture of the Azure Landing Zones. The section includes a list of initiative IDs, their names, descriptions, and the number of policies.

Updates to `docs/wiki/Whats-new.md`:

* Added a note about the missed Q3 timelines and the reasons for the delay. Also, provided information on the major update to the ALZ Policy and the addition of the "Workload Specific Compliance" section to the ALZ Portal accelerator.
* Included a list of new custom initiatives added to support key Azure workloads/services, enhancements to existing initiatives, and new custom policies added for various workloads.
* Updated the "Deploy-MDFC-Config" for Defender for APIs, which now requires a sub plan to be specified.

Updates to ALZ Portal accelerator:

* [`docs/wiki/Whats-new.md`](diffhunk://#diff-dc6403d6507d3ee6a1e907f047aa79fc3a0733256a2fdf5507a64ac754e7ff61R47-R55): Added a new "Workload Specific Compliance" section to the ALZ Portal accelerator. This new section allows users to apply compliance policies to specific workloads like SQL, Storage, etc. These additional compliance controls are often required by highly regulated industries like financial services and healthcare.

Updates to Defender for APIs:

* [`docs/wiki/Whats-new.md`](diffhunk://#diff-dc6403d6507d3ee6a1e907f047aa79fc3a0733256a2fdf5507a64ac754e7ff61R78): Updated the `Deploy-MDFC-Config` for Defender for APIs, which now requires a sub-plan to be specified. The default sub-plan is "P1", and costs will only be incurred once an API has been onboarded to Defender for APIs. Users are advised to review Defender for API plans as they relate to their environment and adjust the sub-plan as needed.

Updates to documentation:

* [`.github/workflows/update-portal.yml`](diffhunk://#diff-7feec2826b1f94812c11d71a139701c78efde8414a85be5dc53306f4aef9620bR83-R85): Added a new step to the workflow to update initiatives in the `jobs:` section.
* [`docs/wiki/ALZ-Policies-Extra.md`](diffhunk://#diff-b6a134a8d04295cc271fa86ab3e814cfc639be84b898364a2f1ca3c28f597725R1-R117): Added a new file to describe additional policies that are not assigned by default or covered in the core ALZ Policies documentation. This file provides guidance on how to handle certain situations and includes a detailed list of policies and initiatives.

### Testing Evidence

![image](https://github.com/Azure/Enterprise-Scale/assets/2101287/1d7e3a5d-8527-41ca-a991-417f83c1a801)

### Testing URLs

#### Azure Public

[![Deploy To Azure](https://learn.microsoft.com/en-us/azure/templates/media/deploy-to-azure.svg)](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FSpringstone%2FEnterprise-Scale%2FFSI%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FSpringstone%2FEnterprise-Scale%2FFSI%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

jtracey93 commented on eslzArm/eslz-portal.json:
```diff
@@ -3845,6 +3845,2334 @@
                 },
+                {
+                    "name": "workloadspecific",
+                    "label": "Workload Specific Compliance",
```
Sorry, we may of discussed this before but it may well of slipped my mind on the outcome.

Do we think that this should be next to and after the regulatory compliance blade, as it feels a little disconnceted and overlapping in the current blade layout/

Springstone replied:
It's not, it's between "Landing Zones..." and "Decommissioned/Sandbox".

jtracey93 commented on src/resources/Microsoft.Authorization/policyDefinitions/Deploy-LogicApp-TLS.json:
```diff
@@ -0,0 +1,99 @@
+        "description": "Periodically, newer versions are released for TLS either due to security flaws, include additional functionality, and enhance speed. Upgrade to the latest TLS version for Function apps to take advantage of security fixes, if any, and/or new functionalities of the latest version.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "App Service",
```
```suggestion
            "category": "Logic Apps",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policyDefinitions/Deny-EH-MINTLS.json:
```diff
@@ -0,0 +1,74 @@
+{
+    "name": "Deny-EH-MINTLS",
```
```suggestion
    "name": "Deny-EH-minTLS",
```
to match others

jtracey93 commented on src/resources/Microsoft.Authorization/policyDefinitions/Deny-LogicApp-Public-Network.json:
```diff
@@ -0,0 +1,70 @@
+        "description": "Disabling public network access improves security by ensuring that the Logic App is not exposed on the public internet. Creating private endpoints can limit exposure of a Logic App. Learn more at: https://aka.ms/app-service-private-endpoint.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "App Service",
```
```suggestion
            "category": "Logic Apps",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policyDefinitions/Deny-LogicApps-Without-Https.json:
```diff
@@ -0,0 +1,70 @@
+        "description": "Use of HTTPS ensures server/service authentication and protects data in transit from network layer eavesdropping attacks.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "App Service",
```
```suggestion
            "category": "Logic Apps",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policyDefinitions/Deny-Service-Endpoints.json:
```diff
@@ -0,0 +1,64 @@
+        "description": "This Policy will deny/audit Service Endpoints on subnets. Service Endpoints allows the network traffic to bypass Network appliances, such as the Azure Firewall.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "App Service",
```
```suggestion
            "category": "Network",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-EncryptTransit.json:
Removing the parameter of `StorageHttpsEnabledEffect` makes this a breaking change, do we need to follow our normal process for this, with a new initative created and deprecate the old one?

Springstone replied:
We're making so many changes here, and this is our breaking change release? I can add a note in what's new

Springstone replied:
Done

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.json:
```diff
@@ -1,364 +1,628 @@
+            },
+            "HealthcareAPIsCMKEffect": {
+                "type": "String",
+                "defaultValue": "audit",
+                "allowedValues": [
+                    "audit",
+                    "disabled"
+                ],
```
casing inconsistency

Springstone replied:
Consistent with the policy (which is case-sensitive :))

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-APIM.json:
No allowed values on parameters

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-CognitiveServices.json:
```diff
@@ -0,0 +1,121 @@
+        "description": "This policy initiative is a group of policies that ensures Cognitive Search is compliant per regulated Landing Zones.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "Cognitive Search",
```
```suggestion
            "category": "Cognitive Services",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-CognitiveServices.json:
Formatting/indents not consistent in file do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-ContainerApps.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-ContainerInstance.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-ContainerInstance.json:
```diff
@@ -0,0 +1,48 @@
+        "description": "This policy initiative is a group of policies that ensures Container Apps is compliant per regulated Landing Zones.",
+        "metadata": {
+            "version": "1.0.0",
+            "category": "Container Instance",
```
```suggestion
            "category": "Container Instances",
```

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-ContainerRegistry.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-CosmosDb.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-DataExplorer.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-DataFactory.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-EventGrid.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-EventHub.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-KeyVault.json:
```diff
@@ -113,7 +113,7 @@
                     "description": "Enter the days before expiration of the certificate when you want to trigger the policy action. For example, to trigger a policy action 90 days before the certificate's expiration, enter '90'."
                 },
-                "defaultValue": 90
+                "defaultValue": 20
```
was this intended to change?

Springstone replied:
I aligned with what FSI has set as a standard. We were never really specific/deliberate in the defaults we set, and 90 days seems overly aggressive (noise easily ignored - 20 days is a concern and requires immediate action).

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-Kubernetes.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-MachineLearning.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-MySQL.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-Network.json:
```diff
@@ -0,0 +1,439 @@
+                "FSI"
+            ]
+        },
+        "parameters": {
```
no allowed values for parameters?

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-AppServices.json:
```diff
@@ -0,0 +1,311 @@
+                "FSI"
+            ]
+        },
+        "parameters": {
```
No allowed values for parameters?

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-OpenAI.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-PostgreSQL.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-ServiceBus.json:
Formatting/indents not consistent in file, do a <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-Storage.json:
```diff
@@ -0,0 +1,372 @@
+                "FSI"
+            ]
+        },
+        "parameters": {
```
No allowed values for parameters?

Springstone replied:
You spotted that did you :)

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-Synapse.json:
```diff
@@ -0,0 +1,172 @@
+                "FSI"
+            ]
+        },
+        "parameters": {
```
No allowed values for parameters?

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Guardrails-Template.json:
was this meant tom be here? I get the idea of a template, but i wouldnt put it in this driectory as we will likely have tools that will walk the directory and this may make it bork as it's not a valid initiative

Springstone replied:
Will drop it... needed it for obvious reasons, but it's done now :)

jtracey93 reviewed (changes requested):
Great work here @Springstone, awesome, awesome work.

Other comments that arent applicable to files:

1. Have we added `metadata.alzVariant` for policy definitions to:
    - Pester Tests
    - Contribution Guides
2. Are we adding tests for all the Deny assignments, into the testing framework that you built?
3. In the new definitions added should the `properties.mode` be set to `all` instead of `indexed` as per https://learn.microsoft.com/azure/governance/policy/concepts/definition-structure-basics#mode
4. Do we need to add the `properties.version` to all our policy definitions now?
6. Address "secure-by-default" definition concerns and update all policies based on outcome etc.

Springstone commented on src/resources/Microsoft.Authorization/policyDefinitions/Audit-PublicIpAddresses-UnusedResourcesCostOptimization.json:
```diff
@@ -40,8 +40,8 @@
-                        "field": "Microsoft.Network/publicIPAddresses/sku.name",
-                        "notEquals": "Basic"
+                        "field": "Microsoft.Network/publicIPAddresses/publicIPAllocationMethod",
+                        "Equals": "Static"
```
Yes, Julian reported that its never compliant and asked to update using this criteria.

Springstone commented on eslzArm/managementGroupTemplates/policyDefinitions/README.md:
```diff
@@ -1,16 +1,16 @@
-# Information relating to `policies.json`
+# Information relating to `policies.json` and `initiatives.json`
```
Definitely not `definitions.bicep`, least of all evils is `policyDefinitions.bicep` but that means all downstream consumers need to make bigger changes.

Springstone commented on docs/wiki/ALZ-Policies-Extra.md:
```diff
@@ -0,0 +1,117 @@
+| Deny-Databricks-NoPublicIp | Deny public IPs for Databricks cluster | |
+| Deny-Databricks-Sku | Deny non-premium Databricks sku | |
+| Deny-Databricks-VirtualNetwork | Deny Databricks workspaces without Vnet injection | |
+| deny-fileservices-insecureauth | File Services with insecure authentication methods should be denied | |
+| deny-fileservices-insecurekerberos | File Services with insecure Kerberos ticket encryption should be denied | |
+| deny-fileservices-insecuresmbchannel | File Services with insecure SMB channel encryption should be denied | |
+| deny-fileservices-insecuresmbversions | File Services with insecure SMB versions should be denied | |
+| deny-machinelearning-aks | Deny AKS cluster creation in Azure Machine Learning | |
+| deny-machinelearning-compute-subnetid | Enforce subnet connectivity for Azure Machine Learning compute clusters and compute instances | |
+| deny-machinelearning-compute-vmsize | Limit allowed vm sizes for Azure Machine Learning compute clusters and compute instances | |
+| deny-machinelearning-computecluster-remoteloginportpublicaccess | Deny public access of Azure Machine Learning clusters via SSH | |
+| deny-machinelearning-computecluster-scale | Enforce scale settings for Azure Machine Learning compute clusters | |
+| deny-machinelearning-hbiworkspace | Enforces high business impact Azure Machine Learning Workspaces | |
+| deny-machinelearning-publicaccesswhenbehindvnet | Deny public access behind vnet to Azure Machine Learning workspace | |
+| deny-machinelearning-publicnetworkaccess | [Deprecated] Azure Machine Learning should have disabled public network access | Deprecated |
+| deny-private-dns-zones | Deny the creation of private DNS | |
+| deny-publicendpoint-mariadb | [Deprecated] Public network access should be disabled for MariaDB | Deprecated |
+| deny-publicip | [Deprecated] Deny the creation of public IP | Deprecated |
+| deny-rdp-from-internet | [Deprecated] RDP access from the Internet should be blocked | Deprecated, replaced by Deny-MgmtPorts-From-Internet |
+| deny-storage-sftp | Storage Accounts with SFTP enabled should be denied | |
+| deny-storageaccount-customdomain | Storage Accounts with custom domains assigned should be denied | |
+| deny-subnet-without-penp | Subnets without Private Endpoint Network Policies enabled should be denied | |
+| deny-subnet-without-udr | Subnets should have a User Defined Route | |
+| deny-udr-with-specific-nexthop | User Defined Routes with 'Next Hop Type' set to 'Internet' or 'VirtualNetworkGateway' should be denied | |
+| deny-vnet-peering | Deny vNet peering | |
+| deny-vnet-peering-to-non-approved-vnets | Deny vNet peering to non-approved vNets | |
+| deploy_vm_availablememory_alert | Deploy VM Available Memory Alert | |
+| deploy_vm_heartbeat_alert_rg | Deploy VM HeartBeat Alert | |
+| deploy-budget | Deploy a default budget on all subscriptions under the assigned scope | |
+| deploy-custom-route-table | Deploy a route table with specific user defined routes | |
+| deploy-ddosprotection | Deploy an Azure DDoS Network Protection | |
+| deploy-firewallpolicy | Deploy Azure Firewall Manager policy in the subscription | |
+| deploy-nsg-flowlogs | [Deprecated] Deploys NSG flow logs and traffic analytics | Deprecated |
+| deploy-nsg-flowlogs-to-la | [Deprecated] Deploys NSG flow logs and traffic analytics to Log Analytics | Deprecated |
+| deploy-sql-tde | [Deprecated] Deploy SQL Database Transparent Data Encryption | Deprecated |
+| deploy-sql-vulnerabilityassessments_20230706 | Deploy SQL Database Vulnerability Assessments | |
+| deploy-vnet-hubspoke | Deploy Virtual Network with peering to the hub | |
+| deploy-windows-domainjoin | Deploy Windows Domain Join Extension with keyvault configuration | |
```
Yes, the script failed me :D

Springstone commented on docs/wiki/ALZ-Policies-Extra.md:
```diff
@@ -0,0 +1,117 @@
+
+| Initiative ID | Name | Description | # of Policies |
+|------------|-------------|-------------|-------------|
+| [Enforce-Guardrails-APIM](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-APIM.html) | Enforce secure-by-default API Management for regulated industries | This policy initiative is a group of policies that ensures API Management is compliant per regulated Landing Zones. | 11 |
+| [Enforce-Guardrails-AppServices](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-AppServices.html) | Enforce secure-by-default App Service for regulated industries | This policy initiative is a group of policies that ensures App Service is compliant per regulated Landing Zones. | 19 |
+| [Enforce-Guardrails-Automation](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Automation.html) | Enforce secure-by-default Automation Account for regulated industries | This policy initiative is a group of policies that ensures Automation Account is compliant per regulated Landing Zones. | 6 |
+| [Enforce-Guardrails-CognitiveServices](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-CognitiveServices.html) | Enforce secure-by-default Cognitive Services for regulated industries | This policy initiative is a group of policies that ensures Cognitive Services is compliant per regulated Landing Zones. | 5 |
+| [Enforce-Guardrails-Compute](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Compute.html) | Enforce secure-by-default Compute for regulated industries | This policy initiative is a group of policies that ensures Compute is compliant per regulated Landing Zones. | 2 |
+| [Enforce-Guardrails-ContainerApps](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-ContainerApps.html) | Enforce secure-by-default Container Apps for regulated industries | This policy initiative is a group of policies that ensures Container Apps is compliant per regulated Landing Zones. | 2 |
+| [Enforce-Guardrails-ContainerInstance](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-ContainerInstance.html) | Enforce secure-by-default Container Instance for regulated industries | This policy initiative is a group of policies that ensures Container Instance is compliant per regulated Landing Zones. | 1 |
+| [Enforce-Guardrails-ContainerRegistry](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-ContainerRegistry.html) | Enforce secure-by-default Container Registry for regulated industries | This policy initiative is a group of policies that ensures Container Registry is compliant per regulated Landing Zones. | 12 |
+| [Enforce-Guardrails-CosmosDb](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-CosmosDb.html) | Enforce secure-by-default Cosmos DB for regulated industries | This policy initiative is a group of policies that ensures Cosmos DB is compliant per regulated Landing Zones. | 6 |
+| [Enforce-Guardrails-DataExplorer](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-DataExplorer.html) | Enforce secure-by-default Data Explorer for regulated industries | This policy initiative is a group of policies that ensures Data Explorer is compliant per regulated Landing Zones. | 4 |
+| [Enforce-Guardrails-DataFactory](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-DataFactory.html) | Enforce secure-by-default Data Factory for regulated industries | This policy initiative is a group of policies that ensures Data Factory is compliant per regulated Landing Zones. | 5 |
+| [Enforce-Guardrails-EventGrid](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-EventGrid.html) | Enforce secure-by-default Event Grid for regulated industries | This policy initiative is a group of policies that ensures Event Grid is compliant per regulated Landing Zones. | 8 |
+| [Enforce-Guardrails-EventHub](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-EventHub.html) | Enforce secure-by-default Event Hub for regulated industries | This policy initiative is a group of policies that ensures Event Hub is compliant per regulated Landing Zones. | 4 |
+| [Enforce-Guardrails-KeyVault-Sup](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-KeyVault-Sup.html) | Enforce secure-by-default Key Vault for regulated industries | This policy initiative is a group of policies that ensures Key Vault is compliant per regulated Landing Zones. This includes additional policies to supplement Enforce-Guardrails-KeyVault, which is assi ...
+| [Enforce-Guardrails-Kubernetes](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Kubernetes.html) | Enforce secure-by-default Kubernetes for regulated industries | This policy initiative is a group of policies that ensures Kubernetes is compliant per regulated Landing Zones. | 16 |
+| [Enforce-Guardrails-MachineLearning](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-MachineLearning.html) | Enforce secure-by-default Machine Learning for regulated industries | This policy initiative is a group of policies that ensures Machine Learning is compliant per regulated Landing Zones. | 5 |
+| [Enforce-Guardrails-MySQL](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-MySQL.html) | Enforce secure-by-default Kubernetes for regulated industries | This policy initiative is a group of policies that ensures Kubernetes is compliant per regulated Landing Zones. | 2 |
+| [Enforce-Guardrails-Network](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Network.html) | Enforce secure-by-default Network and Networking services for regulated industries | This policy initiative is a group of policies that ensures Network and Networking services is compliant per regulated Landing Zones. | 22 |
+| [Enforce-Guardrails-OpenAI](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-OpenAI.html) | Enforce secure-by-default Open AI (Cognitive Services) for regulated industries | This policy initiative is a group of policies that ensures Open AI (Cognitive Services) is compliant per regulated Landing Zones. | 6 |
+| [Enforce-Guardrails-PostgreSQL](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-PostgreSQL.html) | Enforce secure-by-default PostgreSQL for regulated industries | This policy initiative is a group of policies that ensures PostgreSQL is compliant per regulated Landing Zones. | 1 |
+| [Enforce-Guardrails-ServiceBus](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-ServiceBus.html) | Enforce secure-by-default Service Bus for regulated industries | This policy initiative is a group of policies that ensures Service Bus is compliant per regulated Landing Zones. | 4 |
+| [Enforce-Guardrails-SQL](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-SQL.html) | Enforce secure-by-default SQL and SQL Managed Instance for regulated industries | This policy initiative is a group of policies that ensures SQL and SQL Managed Instance is compliant per regulated Landing Zones. | 5 |
+| [Enforce-Guardrails-Storage](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Storage.html) | Enforce secure-by-default Storage for regulated industries | This policy initiative is a group of policies that ensures Storage is compliant per regulated Landing Zones. | 22 |
+| [Enforce-Guardrails-Synapse](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Synapse.html) | Enforce secure-by-default Synapse for regulated industries | This policy initiative is a group of policies that ensures Synapse is compliant per regulated Landing Zones. | 9 |
+| [Enforce-Guardrails-VirtualDesktop](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-VirtualDesktop.html) | Enforce secure-by-default Virtual Desktop for regulated industries | This policy initiative is a group of policies that ensures Virtual Desktop is compliant per regulated Landing Zones. | 2 |
```
Using FSI terminology.

Springstone commented on docs/wiki/ALZ-Policies-Extra.md:
```diff
@@ -0,0 +1,117 @@
+| [Enforce-Guardrails-Synapse](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-Synapse.html) | Enforce secure-by-default Synapse for regulated industries | This policy initiative is a group of policies that ensures Synapse is compliant per regulated Landing Zones. | 9 |
+| [Enforce-Guardrails-VirtualDesktop](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Guardrails-VirtualDesktop.html) | Enforce secure-by-default Virtual Desktop for regulated industries | This policy initiative is a group of policies that ensures Virtual Desktop is compliant per regulated Landing Zones. | 2 |
+
+To support these initiatives we've added the following custom policies:
+
+| Policy ID | Name | Description |
+|------------|-------------|-------------|
+| [Deny-APIM-TLS](https://www.azadvertizer.net/azpolicyadvertizer/Deny-APIM-TLS.html) | API Management services should use TLS version 1.2 | API Management services should use TLS version 1.2 |
+| [Deny-AppGw-Without-Tls](https://www.azadvertizer.net/azpolicyadvertizer/Deny-AppGw-Without-Tls.html) | Application Gateway should be deployed with predefined Microsoft policy that is using TLS version 1.2 | This policy enables you to restrict that Application Gateways is always deployed with predefined Microsoft policy that is using TLS version 1.2 |
+| [Deny-AppService-without-BYOC](https://www.azadvertizer.net/azpolicyadvertizer/Deny-AppService-without-BYOC.html) | App Service certificates must be stored in Key Vault | App Service (including Logic apps and Function apps) must use certificates stored in Key Vault |
+| [Deny-AzFw-Without-Policy](https://www.azadvertizer.net/azpolicyadvertizer/Deny-AzFw-Without-Policy.html) | Azure Firewall should have a default Firewall Policy | This policy denies the creation of Azure Firewall without a default Firewall Policy. |
+| [Deny-CognitiveServices-NetworkAcls](https://www.azadvertizer.net/azpolicyadvertizer/Deny-CognitiveServices-NetworkAcls.html) | Network ACLs should be restricted for Cognitive Services | Azure Cognitive Services should not allow adding individual IPs or virtual network rules to the service-level firewall. Enable this to restrict inbound network access and enforce the usage of private endpoints. ...
+| [Deny-CognitiveServices-Resource-Kinds](https://www.azadvertizer.net/azpolicyadvertizer/Deny-CognitiveServices-Resource-Kinds.html) | Only explicit kinds for Cognitive Services should be allowed | Azure Cognitive Services should only create explicit allowed kinds. |
+| [Deny-CognitiveServices-RestrictOutboundNetworkAccess](https://www.azadvertizer.net/azpolicyadvertizer/Deny-CognitiveServices-RestrictOutboundNetworkAccess.html) | Outbound network access should be restricted for Cognitive Services | Azure Cognitive Services allow restricting outbound network access. Enable this to limit outbound connectivity for the service. |
+| [Deny-EH-MINTLS](https://www.azadvertizer.net/azpolicyadvertizer/Deny-EH-MINTLS.html) | Event Hub namespaces should use a valid TLS version | Event Hub namespaces should use a valid TLS version. |
+| [Deny-EH-Premium-CMK](https://www.azadvertizer.net/azpolicyadvertizer/Deny-EH-Premium-CMK.html) | Event Hub namespaces (Premium) should use a customer-managed key for encryption | Event Hub namespaces (Premium) should use a customer-managed key for encryption. |
+| [Deny-LogicApp-Public-Network](https://www.azadvertizer.net/azpolicyadvertizer/Deny-LogicApp-Public-Network.html) | Logic apps should disable public network access | Disabling public network access improves security by ensuring that the Logic App is not exposed on the public internet. Creating private endpoints can limit exposure of a Logic App. Learn more at: https://aka.ms/app-service-private ...
+| [Deny-LogicApps-Without-Https](https://www.azadvertizer.net/azpolicyadvertizer/Deny-LogicApps-Without-Https.html) | Logic app should only be accessible over HTTPS | Use of HTTPS ensures server/service authentication and protects data in transit from network layer eavesdropping attacks. |
+| [Deny-Service-Endpoints](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Service-Endpoints.html) | Deny or Audit service endpoints on subnets | This Policy will deny/audit Service Endpoints on subnets. Service Endpoints allows the network traffic to bypass Network appliances, such as the Azure Firewall. |
+| [Deny-Storage-ContainerDeleteRetentionPolicy](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-ContainerDeleteRetentionPolicy.html) | Storage Accounts should use a container delete retention policy | Enforce container delete retention policies larger than seven days for storage account. Enable this for increased data loss protection. |
+| [Deny-Storage-CopyScope](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-CopyScope.html) | Allowed Copy scope should be restricted for Storage Accounts | Azure Storage accounts should restrict the allowed copy scope. Enforce this for increased data exfiltration protection. |
+| [Deny-Storage-CorsRules](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-CorsRules.html) | Storage Accounts should restrict CORS rules | Deny CORS rules for storage account for increased data exfiltration protection and endpoint protection. |
+| [Deny-Storage-LocalUser](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-LocalUser.html) | Local users should be restricted for Storage Accounts | Azure Storage accounts should disable local users for features like SFTP. Enforce this for increased data exfiltration protection. |
+| [Deny-Storage-NetworkAclsBypass](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-NetworkAclsBypass.html) | Network ACL bypass option should be restricted for Storage Accounts | Azure Storage accounts should restrict the bypass option for service-level network ACLs. Enforce this for increased data exfiltration protection. |
+| [Deny-Storage-NetworkAclsVirtualNetworkRules](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-NetworkAclsVirtualNetworkRules.html) | Virtual network rules should be restricted for Storage Accounts | Azure Storage accounts should restrict the virtual network service-level network ACLs. Enforce this for increased data exfiltration protection. |
+| [Deny-Storage-ResourceAccessRulesResourceId](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-ResourceAccessRulesResourceId.html) | Resource Access Rules resource IDs should be restricted for Storage Accounts | Azure Storage accounts should restrict the resource access rule for service-level network ACLs to services from a specific Azure subscription. Enforce this for increased data ...
+| [Deny-Storage-ResourceAccessRulesTenantId](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-ResourceAccessRulesTenantId.html) | Resource Access Rules Tenants should be restricted for Storage Accounts | Azure Storage accounts should restrict the resource access rule for service-level network ACLs to service from the same AAD tenant. Enforce this for increased data exfiltration protec ...
+| [Deny-Storage-ServicesEncryption](https://www.azadvertizer.net/azpolicyadvertizer/Deny-Storage-ServicesEncryption.html) | Encryption for storage services should be enforced for Storage Accounts | Azure Storage accounts should enforce encryption for all storage services. Enforce this for increased encryption scope. |
+| [Deploy-LogicApp-TLS](https://www.azadvertizer.net/azpolicyadvertizer/Deploy-LogicApp-TLS.html) | Configure Logic apps to use the latest TLS version | Periodically, newer versions are released for TLS either due to security flaws, include additional functionality, and enhance speed. Upgrade to the latest TLS version for Function apps to take advantage of security fixes, if any, and/or new funct ...
+| [Modify-NSG](https://www.azadvertizer.net/azpolicyadvertizer/Modify-NSG.html) | Enforce specific configuration of Network Security Groups (NSG) | This policy enforces the configuration of Network Security Groups (NSG). |
+| [Modify-UDR](https://www.azadvertizer.net/azpolicyadvertizer/Modify-UDR.html) | Enforce specific configuration of User-Defined Routes (UDR) | This policy enforces the configuration of User-Defined Routes (UDR) within a subnet. |
```
Agreed, and we provide links to the policy for deeper understanding of what the policy actually does. However, doesn't hurt to have an overview of all the policies that ALZ provides (as part of policy sets). Can always improve/expand on this.

Springstone commented:
> Great work here @Springstone, awesome, awesome work.
> 
> Other comments that arent applicable to files:
> 
> 1. Have we added `metadata.alzVariant` for policy definitions to:
>    
>    * Pester Tests
>    * Contribution Guides
> 2. Are we adding tests for all the Deny assignments, into the testing framework that you built?
> 3. In the new definitions added should the `properties.mode` be set to `all` instead of `indexed` as per https://learn.microsoft.com/azure/governance/policy/concepts/definition-structure-basics#mode
> 4. Do we need to add the `properties.version` to all our policy definitions now?
> 5. Address "secure-by-default" definition concerns and update all policies based on outcome etc.

Addressed all the questions here, and created a user story to create additional Pester tests for the new Deny custom policies we're adding in this PR.

jtracey93 commented on docs/wiki/ALZ-Policies-Extra.md:
```diff
@@ -1,117 +1,57 @@
-| deploy-sql-vulnerabilityassessments_20230706 | Deploy SQL Database Vulnerability Assessments | |
-| deploy-vnet-hubspoke | Deploy Virtual Network with peering to the hub | |
-| deploy-windows-domainjoin | Deploy Windows Domain Join Extension with keyvault configuration | |
+| Deny-Appgw-without-Waf | Application Gateway should be deployed with WAF enabled | Use to ensure Application Gateways are deployed with Web Application Firewall enabled |
```
```suggestion
| Deny-Appgw-Without-Waf | Application Gateway should be deployed with WAF enabled | Use to ensure Application Gateways are deployed with Web Application Firewall enabled |
```

Casing?

jtracey93 commented on src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-MDFC-Config.json:
If this is being deprecated why are there any changes apart from the deprecated metadata stuff etc.?

Springstone replied:
A change in a previous PR made breaking changes to this file, which was not intended. I've reverted those changes (Defender for APIs) and made the change in the correct file Deploy-MDFC-Config_20240319.json

jtracey93 reviewed (changes requested):
Just a couple of bits and we should be good
